src/World.h

```diff
@@ -819,6 +822,15 @@ class cWorld :
 		virtual void Execute(void) override;
 	} ;
 
+	struct EntityRemovalQueueItem
+	{
+		EntityRemovalQueueItem(cChunk * a_Chunk, cEntity * a_Entity, bool a_FreeMemory);
+		cChunk * m_Chunk;
+		cEntity * m_Entity;
+		bool m_FreeMemory;
+	};
+
+	std::list<EntityRemovalQueueItem> m_EntityRemovalQueue;
```

Comment:
Is the task guaranteed to be executed just after the mob tick and before the next tick cycle?